packages/core/utils/lib/convert-query-params.js

```diff
@@ -57,38 +85,47 @@ const convertOrderingQueryParams = (ordering) => {
  * Sort query parser
  * @param {string} sortQuery - ex: id:asc,price:desc
  */
-const convertSortQueryParams = (sortQuery) => {
+const convertSortQueryParams = (sortQuery, schema) => {
   if (typeof sortQuery === 'string') {
-    return sortQuery.split(',').map((value) => convertSingleSortQueryParam(value));
+    return sortQuery.split(',').map((value) => convertSingleSortQueryParam(value, schema));
   }
 
   if (Array.isArray(sortQuery)) {
-    return sortQuery.flatMap((sortValue) => convertSortQueryParams(sortValue));
+    return sortQuery.flatMap((sortValue) => convertSortQueryParams(sortValue, schema));
   }
 
   if (_.isPlainObject(sortQuery)) {
-    return convertNestedSortQueryParam(sortQuery);
+    return convertNestedSortQueryParam(sortQuery, schema);
   }
 
   throw new InvalidSortError();
 };
 
-const convertSingleSortQueryParam = (sortQuery) => {
+const convertSingleSortQueryParam = (sortQuery, schema) => {
   // split field and order param with default order to ascending
   const [field, order = 'asc'] = sortQuery.split(':');
 
   if (field.length === 0) {
     throw new Error('Field cannot be empty');
   }
 
+  // skip the check for relational sorts as there is not enough information in the schema.
+  if (!_.get(schema, `attributes.${field}`) && field !== 'id' && !isRelationalSortField(field)) {
+    return {};
+  }
```

Comment:
We support sorting based on [fields from relations](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/entity-service/order-pagination.html#relational-ordering), where you can ask for `/api/<content-type>?sort=<relation>.<relationField>`

This sort would be ignored if not using the `!isRelationalSortField(field)`, and I am wondering If there are other filters that use the dot notation. Do you know of any? @alexandrebodin